### ISSUE TITLE
Introduce Test mode

### ISSUE THREAD

**Issue by DaniRey** [scalacenter/scastie#361]
Currently Scastie provides a worksheet mode. In my opinion it should provide a "test mode" as well.

I would expect that the test mode behaves simliar to executing "test" in sbt.

This would be a huge help in writing executable documentation for testing frameworks.

**Comment by MasseGuillaume** [comment]
It's possible to run them via an entry point:

for example scalatest: http://doc.scalatest.org/1.8/org/scalatest/run$.html

**Comment by DaniRey** [comment]
It absolutely is already possible to run test within Scastie, but I would prefer it to be a first class citizen.

I would suggest to have three modes
- normal
- worksheet
- test

What do you think about this idea?

**Comment by MasseGuillaume** [comment]
Yeah, I think it's a good idea. @julienrf also wanted to add support for `compile` without running.
